File: sdk/examples/typescript/quickstart.ts
// Two-minute TypeScript / Node quickstart for the CasperProver API.
//
// Run (Node 20+):
//     tsx sdk/examples/typescript/quickstart.ts
//     # or
//     ts-node sdk/examples/typescript/quickstart.ts
//
// Env:
//     CP_API_URL   base URL (default http://localhost:9090)
//     CP_API_KEY   optional X-API-Key

const BASE = process.env.CP_API_URL ?? "http://localhost:9090";
const API_KEY = process.env.CP_API_KEY;

const headers: Record<string, string> = { "Content-Type": "application/json" };
if (API_KEY) headers["X-API-Key"] = API_KEY;

async function call<T = unknown>(method: string, path: string, body?: unknown): Promise<T> {
  const resp = await fetch(`${BASE}${path}`, {
    method,
    headers,
    body: body === undefined ? undefined : JSON.stringify(body),
  });
  if (!resp.ok) {
    const text = await resp.text();
    throw new Error(`${method} ${path} -> ${resp.status}: ${text}`);
  }
  const ct = resp.headers.get("content-type") ?? "";
  return ct.includes("json") ? ((await resp.json()) as T) : ((await resp.text()) as unknown as T);
}

function pretty(label: string, obj: unknown): void {
  console.log(`--- ${label} ---`);
  console.log(JSON.stringify(obj, null, 2));
}

async function main(): Promise<void> {
  // 1. Health check.
  const health = await call("GET", "/health");
  pretty("health", health);

  // 2. Submit a proof.
  const proofHash = "beefdead" + Date.now().toString(16).padStart(16, "0");
  const submit = await call<{ proof_id?: string }>("POST", "/proofs", {
    agent_id: "quickstart-ts",
    proof_hash: proofHash,
  });
  pretty("submit_proof", submit);

  const pid = submit.proof_id;
  if (!pid) throw new Error("no proof_id returned");

  // 3. Get.
  const got = await call("GET", `/proofs/${pid}`);
  pretty("get_proof", got);

  // 4. Verify.
  const ver = await call("POST", `/proofs/${pid}/verify`);
  pretty("verify_proof", ver);

  console.log("quickstart OK.");
}

main().catch((err) => {
  console.error("quickstart failed:", err);
  process.exit(1);
});

File: sdk/examples/python/quickstart.py
"""Two-minute Python quickstart for the CasperProver API.

Run:
    python -m sdk.examples.python.quickstart

Requires:
    pip install requests

Env:
    CP_API_URL   base URL (default http://localhost:9090)
    CP_API_KEY   optional X-API-Key
"""
from __future__ import annotations

import json
import os
import sys
import time
from typing import Any

import requests

BASE = os.getenv("CP_API_URL", "http://localhost:9090")
API_KEY = os.getenv("CP_API_KEY")

HEADERS: dict[str, str] = {"Content-Type": "application/json"}
if API_KEY:
    HEADERS["X-API-Key"] = API_KEY


def _pretty(label: str, obj: Any) -> None:
    print(f"--- {label} ---")
    print(json.dumps(obj, indent=2, ensure_ascii=False))


def _call(method: str, path: str, body: Any = None) -> Any:
    url = f"{BASE}{path}"
    resp = requests.request(method, url, headers=HEADERS, json=body, timeout=15)
    if resp.status_code >= 400:
        print(f"error {resp.status_code}: {resp.text}", file=sys.stderr)
        sys.exit(1)
    ct = resp.headers.get("content-type", "")
    return resp.json() if "json" in ct else resp.text


def main() -> None:
    # 1. Health check first — makes failure obvious.
    health = _call("GET", "/health")
    _pretty("health", health)

    # 2. Submit a demo proof.
    proof_hash = "cafebabe" + f"{time.time_ns():016x}"
    submit = _call("POST", "/proofs", {"agent_id": "quickstart-py", "proof_hash": proof_hash})
    _pretty("submit_proof", submit)

    pid = submit.get("proof_id")
    if not pid:
        print("no proof_id returned", file=sys.stderr)
        sys.exit(1)

    # 3. Fetch it.
    got = _call("GET", f"/proofs/{pid}")
    _pretty("get_proof", got)

    # 4. Verify.
    ver = _call("POST", f"/proofs/{pid}/verify")
    _pretty("verify_proof", ver)

    print("quickstart OK.")


if __name__ == "__main__":
    main()
